test/serverExplorer.test.ts

```diff
@@ -0,0 +1,43 @@
+import * as chai from 'chai';
```

Comment:
chai and sinon are not going to work for typescript. What you should do are:
1. mock Odo interface instance to return hierarchy of objects: projects, apps and components
2. pass mocked instance to OpenShiftExplorerDataProvider instance
3. test every method of OpenShiftExplorerDataProvider instance returns right number of objects mocked on step 1

Reply:
I have no such experience. The client project uses both for typescript, you just need the respective types.
And what exactly does Odo and OpenShift have to do with the adapter extension right now?

Reply:
@dgolovin I think you meant for server-adpaters PR and not for openshift-tools one.